pimd/pim_register.c

```diff
@@ -131,32 +156,58 @@ int pim_register_stop_recv(struct interface *ifp, uint8_t *buf, int buf_size)
 	pim_parse_addr_ucast(&source, buf, buf_size);
 	sg.src = source.u.prefix4;
 
+	rp = RP(pim_ifp->pim, sg.grp);
+
+	if (sg.src.s_addr == rp->rpf_addr.u.prefix4.s_addr)
+		sg.src.s_addr = INADDR_ANY;
+
 	upstream = pim_upstream_find(pim, &sg);
-	if (!upstream) {
+
+	if ((!upstream) && (sg.src.s_addr != INADDR_ANY))
 		return 0;
-	}
 
-	if (PIM_DEBUG_PIM_REG)
```

Comment:
I'd put this debug back in fact I would put it before the return right above on line 167.  Let people know you received a register stop with a particular multicast (S,G) address.

Reply:
Agreed will put it back for non null upstream.

Reply:
put it back for the S,G received.

Reply:
Done